applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -230,6 +234,239 @@ void CopyNodalSolutionStepVariablesList(
 
     KRATOS_CATCH("");
 }
+
+template <typename TDataType>
+void AssignConditionVariableValuesToNodes(ModelPart& rModelPart,
+                                          const Variable<TDataType>& rVariable,
+                                          const Flags& rFlag,
+                                          const bool FlagValue)
+{
+    auto& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetHistoricalVariableToZero(rVariable, r_nodes);
+
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+    {
+        auto& r_cond = *(rModelPart.ConditionsBegin() + i_cond);
+        if (r_cond.Is(rFlag) == FlagValue)
+        {
+            const int number_of_nodes = r_cond.GetGeometry().PointsNumber();
+            const auto& r_normal = r_cond.GetValue(rVariable);
+            for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+            {
+                auto& r_node = r_cond.GetGeometry()[i_node];
+                r_node.SetLock();
+                r_node.FastGetSolutionStepValue(rVariable) +=
+                    r_normal * (1.0 / static_cast<double>(number_of_nodes));
+                r_node.UnSetLock();
+            }
+        }
+    }
+
+    rModelPart.GetCommunicator().AssembleCurrentData(rVariable);
+}
+
+void AddAnalysisStep(ModelPart& rModelPart, const std::string& rStepName)
+{
+    auto& r_process_info = rModelPart.GetProcessInfo();
+    if (!r_process_info.Has(ANALYSIS_STEPS))
+    {
+        r_process_info.SetValue(ANALYSIS_STEPS, std::vector<std::string>());
+    }
+    r_process_info[ANALYSIS_STEPS].push_back(rStepName);
+}
+
+bool IsAnalysisStepCompleted(const ModelPart& rModelPart, const std::string& rStepName)
+{
+    const auto& r_process_info = rModelPart.GetProcessInfo();
+    if (r_process_info.Has(ANALYSIS_STEPS))
+    {
+        const std::vector<std::string>& r_steps = r_process_info[ANALYSIS_STEPS];
```

Comment:
What are these `ANALYSIS_STEPS`? Maybe makes sense to do something as in the FS strategy to make it simpler. Maybe an `enum`?

Reply:
These are not exactly what we have FS strategy steps. These analysis steps are used to avoid double calculating stuff. 
Example:
In the case of turbulence models, i need to calculate wall normals. In the case of incompressible potential flow also I need to calculate wall normals. So If some one uses turbulence models with incompressible potential flow solver as initializer, this will avoid calculating wall normals twice. Wall normal calculation is not necessary always such as in pure vms_monolithc solvers or fractional step solvers.